internal/xds/types/resourceversiontable.go

```diff
@@ -68,7 +74,60 @@ func (t *ResourceVersionTable) GetXdsResources() XdsResources {
 	return t.XdsResources
 }
 
-func (t *ResourceVersionTable) AddXdsResource(rType resourcev3.Type, xdsResource types.Resource) {
+func (t *ResourceVersionTable) AddXdsResource(rType resourcev3.Type, xdsResource types.Resource) error {
+	// Perform type switch to handle different types of xdsResource
+	switch rType {
+	case resourcev3.ListenerType:
+		// Handle Type specific operations
+		if resourceOfType, ok := xdsResource.(*listenerv3.Listener); ok {
+			if err := resourceOfType.ValidateAll(); err != nil {
+				return fmt.Errorf("validation failed for xds resource %+v, err:%v", xdsResource, err)
+			}
+		} else {
+			return fmt.Errorf("failed to cast xds resource %+v to Listener type", xdsResource)
+		}
+	case resourcev3.RouteType:
+		// Handle Type specific operations
+		if resourceOfType, ok := xdsResource.(*routev3.RouteConfiguration); ok {
+			if err := resourceOfType.ValidateAll(); err != nil {
+				return fmt.Errorf("validation failed for xds resource %+v, err:%v", xdsResource, err)
+			}
+		} else {
+			return fmt.Errorf("failed to cast xds resource %+v to RouteConfiguration type", xdsResource)
+		}
+
+	case resourcev3.SecretType:
+		// Handle specific operations
+		if resourceOfType, ok := xdsResource.(*tlsv3.Secret); ok {
+			if err := resourceOfType.ValidateAll(); err != nil {
+				return fmt.Errorf("validation failed for xds resource %+v, err:%v", xdsResource, err)
+			}
+		} else {
+			return fmt.Errorf("failed to cast xds resource %+v to RouteConfiguration type", xdsResource)
+		}
+
+	case resourcev3.EndpointType:
+		// TBD - ValidateAll() breaks existing test internal/cmd/egctl/translate_test
```

Comment:
thanks for catching this, lets raise a issue to track this

Reply:
Sure, I have opened https://github.com/envoyproxy/gateway/issues/1742. Here is the PR to fix the issue https://github.com/envoyproxy/gateway/pull/1758